### ISSUE TITLE
Very frequent random crash

### ISSUE THREAD

**Issue by sadaiyandi-exalens** [questdb/questdb#2019]
### Describe the bug

We are seeing the very frequent crash. there is no particular sequence or with certain data. the crash is really random.

 A fatal error has been detected by the Java Runtime Environment:
 SIGSEGV (0xb) at pc=0x00007f88f8bfd930, pid=548526, tid=548553
JRE version: OpenJDK Runtime Environment AdoptOpenJDK (11.0.9.1+1) (build 11.0.9.1+1)
Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK (11.0.9.1+1, mixed mode, tiered, compressed oops, parallel gc, linux-amd64)
 Problematic frame:
 J 2408 c2 io.questdb.cairo.vm.api.MemoryCR.getLong(J)J io.questdb@6.2 (44 bytes) @ 0x00007f88f8bfd930 [0x00007f88f8bfd8c0+0x0000000000000070]

 No core dump will be written. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again

stdout-2022-04-07T22-34-26.txt](https://github.com/questdb/questdb/files/8446400/stdout-2022-04-07T22-34-26.txt)
[hs_err_pid548526.log](https://github.com/questdb/questdb/files/8446421/hs_err_pid548526.log)

### To reproduce

There is no particular step to reproduce the issue. issues are seen very often with our test case environment

### Expected Behavior

questDB running continuously with out interruption

### Environment

```markdown
- **QuestDB version**:6.2
- **OS**: ubuntu 20.04
- **Browser**:
```

**Comment by ideoma** [comment]
Would be good if you try 6.2.1, few bugs were fixed there.
Also would be good if you can post `hs_err_pidXXXX.log` file of the crash, it's created in the working directory of the process.

**Comment by sadaiyandi-exalens** [comment]
sure. I will try it in 6.2.1
[hs_err_pid563208.log](https://github.com/questdb/questdb/files/8447211/hs_err_pid563208.log)
[hs_err_pid559837.log](https://github.com/questdb/questdb/files/8447212/hs_err_pid559837.log)
[hs_err_pid558993.log](https://github.com/questdb/questdb/files/8447213/hs_err_pid558993.log)

**Comment by sadaiyandi-exalens** [comment]
@ideoma I don't see the crash in 6.2.1. I will update here if I see it in further testing. Thank you so much for your support

**Comment by bziobrowski** [comment]
@sadaiyandi-exalens  Did you experience a crash since switching to 6.2.1 ?

**Comment by sadaiyandi-exalens** [comment]
[stdout-2022-04-14T11-58-44.txt](https://github.com/questdb/questdb/files/8490219/stdout-2022-04-14T11-58-44.txt)

**Comment by ideoma** [comment]
Will it be possible to share code which leads to the crash?
If you could log into a file the data you send to via ILP before the crash that would also help to reproduce.

**Comment by sadaiyandi-exalens** [comment]
@ideoma please find the https://github.com/sadaiyandi-exalens/testdataset/blob/main/DS1-TB11-R79.tar.xz data, column name which has dot are replaced with an underscore.
@bluestreak01 as you suggested we created a table upfront still we are seeing the crash but it reduces the occurrence of the crash and also additionally it's created a new issue.

info: line_proto is a list of strings (ILP)

    self._sock.sendall(f"{''.join(line_proto)}".encode())
BrokenPipeError: [Errno 32] Broken pipe

    self._sock.sendall(f"{''.join(line_proto)}".encode())
ConnectionResetError: [Errno 104] Connection reset by peer
[stdout-2022-04-17T18-41-34.txt](https://github.com/questdb/questdb/files/8502353/stdout-2022-04-17T18-41-34.txt) 

Please let me know if there is a way to avoid this issue, it would be very helpful.

**Comment by bluestreak01** [comment]
The information you provided is very helpful, thank you! The "connection reset" is server disconnecting client in case of malformed ILP text. If you search the log for ` E `, e.g  <space>E<space> you should see type cast errors:

```
2022-04-17T16:41:49.179894Z E i.q.c.l.t.LineTcpConnectionContext [1492] could not process line data [table=files, msg=cast error for line protocol string [columnIndex=3, columnType=DOUBLE], errno=0]
```

Please lets us have a sample ILP line you're producing, perhaps we can help you spot issues

**Comment by sadaiyandi-exalens** [comment]
@bluestreak01 thanks for the suggestion. We are able to fix most of the connection reset. still, we are seeing the connection reset issue. I couldn't make out the root cause of the issue from the log 
[stdout-2022-04-18T09-15-00.txt](https://github.com/questdb/questdb/files/8512846/stdout-2022-04-18T09-15-00.txt)
[hs_err_pid3002.log](https://github.com/questdb/questdb/files/8512847/hs_err_pid3002.log)
.

**Comment by bluestreak01** [comment]
It does seem as if you still adding columns on the fly with ILP. If you grep for ` ADDED`  in the log, you should see that

**Comment by bluestreak01** [comment]
We can reliably reproduce the crash with information from your logs. The workaround seems to provide table with designated timestamp, for example:

```
CREATE TABLE IF NOT EXISTS conn(timestamp TIMESTAMP,ts DOUBLE, uid STRING, src_ip STRING, src_port INT , dst_ip STRING, dst_port INT,is_src_private BOOLEAN, is_dst_private BOOLEAN, ts_update TIMESTAMP , proto STRING, service STRING,duration DOUBLE, orig_bytes LONG, resp_bytes LONG, conn_state STRING , missed_bytes LONG, history STRING, orig_pkts LONG, orig_ip_bytes LONG, resp_pkts LONG, resp_ip_bytes LONG, tunnel_parents STRING, src_mac STRING , dst_mac STRING , vlan LONG, inner_vlan LONG, speculative_service STRING) timestamp (timestamp) partition by day;
```

**Comment by glasstiger** [comment]
Pushed a fix for the issue in the above pull request.
Will be included in version 6.3.
Can provide a patched 6.2.1 version too on request.
Please, ping me if you need it.

**Comment by glasstiger** [comment]
Please, also note that this change is to make sure QuestDB is not crashing when the type of the dynamically added field is mutated.
QuestDB will drop such lines with a type cast error printed to the log.
You should also investigate on your side and make sure that data ingested into QuestDB consistently uses data types.
The description of the pull request above provides more detail and an example of what to look for.
